packages/app-mining/src/components/ClaimAll.tsx

```diff
@@ -96,13 +123,21 @@ const ClaimAll = (props: BlockProps) => {
   const claimableStakePoolPids = useMemo<number[]>(() => {
     return data?.findManyStakePools.map((stakePool) => stakePool.pid) || []
   }, [data])
+  const [confirmLock, setConfirmLock] = useState(false)
 
-  const onConfirm = () => {
-    waitSignAndSend(extrinsic, (status) => {
-      if (status.isReady) {
-        closeModal()
-      }
-    })
+  const onConfirm = async () => {
+    setConfirmLock(true)
+    try {
+      await waitSignAndSend(extrinsic, (status) => {
+        if (status.isReady) {
+          closeModal()
+        }
+      })
+    } catch (err) {
+      setConfirmLock(false)
+    } finally {
+      setConfirmLock(false)
```

Comment:
It seems that there is no need to write `setConfirmLock(false)` twice?

Reply:
have deleted it already.